### PR TITLE
Add types for WebGPU

### PR DESCRIPTION
**Pull request by willeastcott**
Add IntelliSense support for WebGPU types:

image

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by mvaligursky** [comment]
I tried adding those before, fingers crossed you get it to work. 
We'll then need to remove these I think
https://github.com/playcanvas/engine/blob/f1770e4979f326158d66fb948c3e290116d19291/package.json#L37-L57

**Comment by willeastcott** [comment]
@mvaligursky Seems to work. Can you maybe verify for yourself using this branch?

**Review by mvaligursky — APPROVED** [review]
Agreed, seems to work great! I added some types back in, will do the rest in a separate PR when this is merged.

File: package.json [hunk 1]
```diff
@@ -34,26 +34,10 @@
       "Ammo": "readonly",
       "earcut": "readonly",
       "glslang": "readonly",
-      "GPUAdapter": "readonly",
-      "GPUBindGroup": "readonly",
-      "GPUBindGroupDescriptor": "readonly",
-      "GPUBindGroupLayout": "readonly",
-      "GPUBlendState": "readonly",
-      "GPUBuffer": "readonly",
       "GPUBufferUsage": "readonly",
```
Comment (mvaligursky) [inline package.json:37]: these 4 should be removed as well I think

Comment (willeastcott) [reply package.json:37]: The linter complains if you remove those 4.

**Pull request merged**